File: cli/joytrunk/gateway/lib/paths.js
/**
 * JoyTrunk 路径约定：与 cli §4.2 一致。
 * Windows: %USERPROFILE%\.joytrunk，Linux/macOS: ~/.joytrunk
 */

const path = require('path');
const os = require('os');

function getJoytrunkRoot() {
  if (process.env.JOYTRUNK_ROOT) {
    return path.resolve(process.env.JOYTRUNK_ROOT);
  }
  return path.join(os.homedir(), '.joytrunk');
}

function getConfigPath() {
  return path.join(getJoytrunkRoot(), 'config.json');
}

function getWorkspaceRoot() {
  return path.join(getJoytrunkRoot(), 'workspace');
}

function getEmployeeDir(employeeId) {
  return path.join(getWorkspaceRoot(), 'employees', employeeId);
}

/** 员工级 config.json 路径（可覆盖主 config）；无则继承主配置 */
function getEmployeeConfigPath(employeeId) {
  return path.join(getEmployeeDir(employeeId), 'config.json');
}

/** 负责人级共享记忆目录：workspace/memory/，所有员工可见 */
function getSharedMemoryDir() {
  return path.join(getWorkspaceRoot(), 'memory');
}

/** 负责人级共享技能目录：workspace/skills/，所有员工默认可见 */
function getSharedSkillsDir() {
  return path.join(getWorkspaceRoot(), 'skills');
}

module.exports = {
  getJoytrunkRoot,
  getConfigPath,
  getWorkspaceRoot,
  getEmployeeDir,
  getEmployeeConfigPath,
  getSharedMemoryDir,
  getSharedSkillsDir,
};

File: cli/joytrunk/paths.py
"""JoyTrunk 路径：平台无关的根目录与工作区路径。"""

import os
from pathlib import Path


def get_cli_root() -> Path:
    """CLI 项目根目录（joytrunk 包所在目录的上一级）。开发时即 cli/，用于解析 cli/.env。"""
    return Path(__file__).resolve().parent.parent


def get_joytrunk_root() -> Path:
    """JoyTrunk 本地根目录。可通过环境变量 JOYTRUNK_ROOT 覆盖（测试用）。"""
    if os.environ.get("JOYTRUNK_ROOT"):
        return Path(os.environ["JOYTRUNK_ROOT"]).resolve()
    return Path.home() / ".joytrunk"


def get_config_path() -> Path:
    """配置文件路径：~/.joytrunk/config.json"""
    return get_joytrunk_root() / "config.json"


def get_workspace_root() -> Path:
    """工作区根目录：~/.joytrunk/workspace"""
    return get_joytrunk_root() / "workspace"


def get_workspace_skills() -> Path:
    """团队共享技能目录：~/.joytrunk/workspace/skills"""
    return get_workspace_root() / "skills"


def get_workspace_memory() -> Path:
    """团队共享记忆目录：~/.joytrunk/workspace/memory"""
    return get_workspace_root() / "memory"


def get_employee_dir(employee_id: str) -> Path:
    """某员工的 workspace 子目录：~/.joytrunk/workspace/employees/<employee_id>"""
    return get_workspace_root() / "employees" / employee_id
